src/main/java/duke/exceptions/TaskOutOfRangeException.java

```diff
@@ -0,0 +1,23 @@
+package duke.exceptions;
+
+/**
+ * Represents an exception due to invalid task number input during 
+ * done and delete duke.commands.
+ */
+public class TaskOutOfRangeException extends DukeException {
+    /**
+     * Constructor for <code>TaskOutOfRangeException</code>
+     *
+     * @param maxIndex number of duke.tasks a user has
+     */
+    public TaskOutOfRangeException(int maxIndex) {
+        super(maxIndex == 1 
+                ? "Oops! No such task exists! You only have one task.\n"
+                :"Oops! No such task exists\nPlease use a number from 1 to " 
```

Comment:
Just another nitpick with spacing.
```suggestion
                : "Oops! No such task exists\nPlease use a number from 1 to " 
```